lib/webapp.js

```diff
@@ -233,21 +233,55 @@ const staticHandlers = {
         but this should be enforced somehow in the future.
       */
       if (!type) {
-        do404(req.url, res, "A plugin type must be specified");
-        return;
+        type = "all"
       }
       const acceptLanguage = 
         translationUtils.getAcceptLanguageFromCookies(req.cookies) || req.headers['accept-language'] || '';
       const pluginDefs = plugins.map(p => p.exportTranslatedDef(acceptLanguage));
+      let pluginId;
+      let pluginLocation;
+      let filteredPluginDefs = []
+      let allowedPlugins;
+      if (options.serverConfig.dataserviceAuthentication.rbac) {
+        for (let plugin of plugins) {
+          if (plugin.pluginType === "bootstrap") {
+            pluginId = plugin.identifier
+            pluginLocation = plugins.location
+            break;
+          }
+        }
+        if (req.username) {
+          allowedPlugins = configService.getAllowedPlugins(options, req.username, pluginId, pluginLocation);
+        } else { 
+          allowedPlugins = configService.getAllowedPlugins(options, undefined, pluginId, pluginLocation);
```

Comment:
You don't need an if-else because if !req.username, req.username = undefined|null